controllers/volumereplicationgroup_controller.go

```diff
@@ -127,6 +128,7 @@ func pvcPredicateFunc() predicate.Funcs {
 
 			log.Info("Update event for PersistentVolumeClaim")
 
+			// TODO: If finalizers change then deep equal of spec fails to catch it, we may want more conditions here
```

Comment:
Just trying to understand more about this comment. What do you mean by finalizers change? Changing the name of the finalizer in future? Or is it different finalizers being applied at different points in time (i.e. protecting first and then adding protected finalizer followed by removal of protecting finalizer) ?

Reply:
When a we move VRG to `Secondary` we wait for PVCs to be deleted and not in use. The former is part of the `metadata.deletionTimestamp` and the latter is part of `metadata.finalizers` section in the PVC.

As we compare only `spec` section we miss these changes and not trigger a reconcile. The reconcile is actually triggered as we have requeued, so no loss a such, but we take action at the resync interval rather than as soon as the PVC changes its states as above.

So, we could add further conditions here to trigger a reconcile, based on the specific sections comparison as well.